### PR TITLE
Groups list branch

### PR DESCRIPTION
**Pull request by patichandana**
group component preliminary version implemented

### CODE REVIEW

File: src/App.jsx
Comment (anandubajith) [inline src/App.jsx:1]: revert?

File: src/components/Common
Comment (anandubajith) [inline src/components/Common:1]: ?

File: src/index.css [hunk 1]
```diff
@@ -11,3 +18,35 @@ code {
...
+  display: flex;
+  flex-direction: column;
+  justify-content: start;
+}
+
+.first {
+  display:flex;
+  justify-content: space-around;
+  padding: 10px;
+}
```
Comment (anandubajith) [inline src/index.css]: Is this needed? or we moved to using `styled-components`

File: src/index.jsx
Comment (anandubajith) [inline src/index.jsx:1]: revert?

File: src/pages/Group.jsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+`;
+
+const Debt = styled("div")`
+  display: flex;
+  padding: 8px 10px;
+
+  .debt-text {
+    font-size: 15px;
+    color: #7a7a7a;
+    // font-weight: 700;
```
Comment (anandubajith) [inline src/pages/Group.jsx]: remove?

File: src/pages/Group.jsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                <Typography className="debt-text">
+                  {debt.personName} owes you {debt.currency}
+                  {debt.amount}
+                </Typography>
+              ) : (
+                <Typography className="debt-text">
+                  you owe {debt.personName} {debt.currency}
+                  {debt.amount}
+                </Typography>
+              )}
```
Comment (anandubajith) [inline src/pages/Group.jsx]: refactor?
```suggestion
            
                <Typography className="debt-text">
                 { generateMessage(debt) }
                </Typography>

```

File: src/pages/Group.jsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            </Debt>
+          );
+        })}
+      </Balances>
+
+      <PersonBalance balanceAmount={totalBalance}>
+        <Typography className="balance-text">{totalBalanceText}</Typography>
+        <Typography className="balance-amount">{currency}{totalBalance}</Typography>
```
Comment (anandubajith) [inline src/pages/Group.jsx]:
```suggestion
        {currency}{totalBalance}
```

rely on [NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) instead ?

File: src/pages/Group.jsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      </PersonBalance>
+    </GroupContainer>
+  );
+}
+
+function Debts({}) {}
```
Comment (anandubajith) [inline src/pages/Group.jsx]: Is this needed?

File: src/pages/Group.jsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+  console.log(debtsList);
+  let totalBalanceText;
+
+  if (totalBalance == 0) {
+    totalBalanceText = "Settled up";
+  } else if (totalBalance > 0) {
+    totalBalanceText = "You are owed";
+  } else if (totalBalance < 0) {
+    totalBalanceText = "You owe";
+  }
```
Comment (anandubajith) [inline src/pages/Group.jsx]: nit: move to function?

File: src/App.jsx [hunk 7]
```diff
@@ -1,28 +1,59 @@
...
+  //   <div className="App gradiant_background">
+  //     {loadComponent}
+  //   </div>
+  // );
+
+  const debts = [
+    {
+      "personName": "Riya",
+      "amount": "100",
+      'currency': "$"
```
Comment (anandubajith) [inline src/App.jsx]: - favour currency to take in `USD` so that localisation specific NumberFormatter can be used?

File: src/App.jsx [hunk 8]
```diff
@@ -1,28 +1,59 @@
...
+    },{
+      "personName": "Wade",
+      "amount": "100",
+      'currency': "$"
+    }
+  ]
+
+  return  <>
+  <Group debtsList={debts} totalBalance="200" currency="$"></Group>
```
Comment (anandubajith) [inline src/App.jsx]: - Do we need the totalBalance prop, currency props? or can they be inferred from debts?
- Will the group support having debts in multiple currencies? or will a group be fixed to a single currency?
  -  If a group can only have one currency, do we need the currency in debts?

File: src/pages/Group.jsx (hunk 2 above)
Comment (patichandana) [reply src/pages/Group.jsx]: you meant the styles to be placed in another js file?

Comment (anandubajith) [reply src/pages/Group.jsx]: the comment

File: src/components/Common
Comment (anandubajith) [inline src/components/Common:1]: Revert?

File: src/pages/Group.jsx [hunk 9]
```diff
@@ -0,0 +1,176 @@
+import { Avatar, Typography, Divider } from "@mui/material";
+import { styled } from "@mui/material/styles";
+// import CakeTwoToneIcon from '@mui/icons-material/CakeTwoToneIcon';
```
Comment (anandubajith) [inline src/pages/Group.jsx]: Remove?

File: src/pages/Group.jsx [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+      props.balanceAmount > 0
+        ? "#027927"
+        : props.balanceAmount == 0
+        ? "#f5f1e8"
+        : "#b31c3a"};
+    font-weight: 700;
+  }
+`;
+
+/* Group class default export  */
```
Comment (anandubajith) [inline src/pages/Group.jsx]: Why?
image

File: src/pages/Group.jsx [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+export default function Group({ groupData, setPage, setParams }) {
+  console.log("chandana: ", groupData);
```
Comment (anandubajith) [inline src/pages/Group.jsx]: Remove?

File: src/components/Common
Comment (anandubajith) [inline src/components/Common:1]: Empty file still not removed?

File: src/styles/Group.styles.js [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+
+export {
+    GroupContainer,
+    GroupDetails,
+    GroupAvatar,
+    Balances,
+    Debt,
+    PersonBalance
+ };
```
Comment (anandubajith) [inline src/styles/Group.styles.js]: Use named exports instead?

File: src/styles/Group.styles.js [hunk 13]
```diff
@@ -0,0 +1,107 @@
+import { styled } from "@mui/material/styles";
+import { Avatar, Typography, Divider } from "@mui/material";
+
+
+/* custom styled components */
+const GroupContainer = styled("div")`
```
Comment (anandubajith) [inline src/styles/Group.styles.js]:
```suggestion
export const GroupContainer = styled("div")`
```

**Review by anandubajith — APPROVED** [review]

**Pull request merged**